hathor/metrics.py

```diff
@@ -57,135 +57,80 @@ class PeerConnectionMetrics:
-
-        # Blocks count in the network
-        self.blocks = 0
-
-        # Height of the best chain of the network
-        self.best_block_height = 0
-
-        # Hash rate of the network
-        self.hash_rate = 0.0
-
-        # Peers connected
-        self.peers = 0
-
-        # Length of the tx deque
-        self.weight_tx_deque_len = 60
-
-        # Length of the block deque
-        self.weight_block_deque_len = 450
-
         # Stores caculated tx weights saved in tx storage
-        self.weight_tx_deque = deque(maxlen=self.weight_tx_deque_len)
+        self.weight_tx_deque: Deque[WeightValue] = deque(maxlen=self.weight_tx_deque_len)
 
         # Stores caculated block weights saved in tx storage
-        self.weight_block_deque = deque(maxlen=self.weight_block_deque_len)
-
-        self.avg_time_between_blocks = avg_time_between_blocks
-
-        self.pubsub = pubsub
-
-        self.tx_storage = tx_storage or TransactionMemoryStorage()
+        self.weight_block_deque: Deque[WeightValue] = deque(maxlen=self.weight_block_deque_len)
 
-        if reactor is None:
+        if self.reactor is None:
             from hathor.util import reactor as twisted_reactor
-            reactor = twisted_reactor
-        self.reactor = reactor
-
-        # If metric capture data is running
-        self.is_running = False
-
-        # Coefficient of exponential calculus
-        self.exponential_alfa = 0.7
-
-        # Time between method call to store hash count
-        self.tx_hash_store_interval = 1
-        self.block_hash_store_interval = 1
-
-        # Websocket data stored
-        self.websocket_connections = 0
-        self.subscribed_addresses = 0
-
-        # Websocket factory
-        self.websocket_factory = None
-
-        # weight of the head of the best blockchain
-        self.best_block_weight = 0
-
-        # Stratum data
-        self.completed_jobs = 0
-        self.blocks_found = 0
-        self.estimated_hash_rate = 0
-
-        # Stratum factory
-        self.stratum_factory = None
-
-        # Peer Connection data
-        self.connections = connections
-        self.peer_connection_metrics: List[PeerConnectionMetrics] = []
-
-        # TxCache Data
-        self.transaction_cache_hits = 0
-        self.transaction_cache_misses = 0
-
-        # Send-token timeouts counter
-        self.send_token_timeouts = 0
-
-        # Time between method call to collect data
-        self.collect_data_interval = settings.METRICS_COLLECT_DATA_INTERVAL
+            self.reactor = twisted_reactor
 
         # A timer to periodically collect data
         self._lc_collect_data = LoopingCall(self._collect_data)
-        self._lc_collect_data.clock = reactor
+        self._lc_collect_data.clock = self.reactor
+
+        # The number of blocks interval to control periodic collection of RocksDB data
+        # We try to use it if possible, instead of the time interval, because it's better to make sure we update the
```

Comment:
As far as I understood, we are always using it. Is that right?

Reply:
Yes, fixed in https://github.com/HathorNetwork/hathor-core/pull/485/commits/0cc50d54b6cc541c78340b92480db3e64a40b8dc